docs/core/dependency-loading/understanding-assemblyloadcontext.md

```diff
@@ -0,0 +1,81 @@
+  - The exception is that the initial `null` response could be followed by a later non-`null` response.
+  - For managed assemblies in particular, we're creating a <xref:System.Reflection.AssemblyName.Name?displayProperty=nameWithType> to <xref:System.Reflection.Assembly> cache. There can be exactly one <xref:System.Reflection.AssemblyName.Name?displayProperty=nameWithType> in the cache.
+- **Typically don't throw**.  It's expected that these functions return `null` rather than throw when unable to find the requested dependency. Throwing should be restricted to unexpected errors like a corrupt assembly or out of memory.
+- **Avoid recursion**. Beware that these functions and handlers are implementing the loading rules for locating dependencies. Your implementation shouldn't call APIs that trigger recursion. Your code should typically be calling **AssemblyLoadContext** load functions that require a specific path or memory reference argument.
+- **Load into the correct AssemblyLoadContext**. The choice of where to load dependencies is application-specific.  The choice is implemented by these events and functions. When your code calls **AssemblyLoadContext** load-by-path functions call them on the instance where you want the code loaded. Sometime returning `null` and letting the <xref:System.Runtime.Loader.AssemblyLoadContext.Default?displayProperty=nameWithType> handle the load may be the simplest option.
+
+## How are dynamic dependencies isolated?
+
+One of the primary purposes of exposing the <xref:System.Runtime.Loader.AssemblyLoadContext> is to allow for isolation of dependencies.
+
+The general static dependency model requires a one to one mapping of name to dependency instance.
+
+For dynamic dependencies, there's no guarantee that another static or dynamic dependency hasn't already loaded a dependency of the same name.
+
+Each <xref:System.Runtime.Loader.AssemblyLoadContext> instance represents a unique scope for mapping names.
+
+There's no binary isolation between these dependencies. They're only isolated by not finding each other by name.
+
+## How are dependencies shared?
+
+Dependencies can easily be shared between <xref:System.Runtime.Loader.AssemblyLoadContext> instances. The general model is for one <xref:System.Runtime.Loader.AssemblyLoadContext> to load a dependency.  The other share the dependency by adding the dependency instance to their cache.
+
+This sharing is required of the runtime assemblies. These assemblies can only be loaded into the <xref:System.Runtime.Loader.AssemblyLoadContext.Default?displayProperty=nameWithType>. The same is required for frameworks like `ASP.NET`, `WPF`, or `WinForms`.
+
+It's recommended that shared dependencies should be loaded into <xref:System.Runtime.Loader.AssemblyLoadContext.Default?displayProperty=nameWithType>.
```

Comment:
Maybe we should describe why is this recommended (the sharing is semi-automatic via the fact that fallback to Default is part of the runtime's algorithm)